### DISCUSSION TITLE
Linux lib dependencies

### DISCUSSION THREAD

**Discussion by kwalker99** [apohl79/audiogridder#291] [Help] [answered]
Hi Guys
I am trying to scan for the vst and vst3 plugin in Bitwig 3.1.2 on Ubuntu Linux 18.04. Bitwig shows that the libavcodecs are not found:

```
com.bitwig.flt.library.metadata.reader.exception.CouldNotReadMetadataException: could not read metadata: Failed to load VST 2 plug-in /home/kevin/.vst/audiogridder/AudioGridder.so: libavcodec.so.58: cannot open shared object file: No such file or directory
```
```
com.bitwig.flt.library.metadata.reader.exception.CouldNotReadMetadataException: could not read metadata: Failed to load VST 3 plug-in /home/kevin/.vst3/audiogridder/AudioGridder.vst3: libavcodec.so.58: cannot open shared object file: No such file or directory
```
It looks as if you have a lot of dynamic lib dependencies, a quick ldd at the cli reveals that my glibc is older and I am missing:
```
libavcodec.so.58 => not found
libavutil.so.56 => not found
libswscale.so.5 => not found
```

Please confirm which version of Linux you compiled the vst plugin on? Probably worth including this information under the compatibility section on the website, I expect I will have to upgrade my distro...

Kind regards

KW

**Comment by apohl79** [comment, marked as answer]
Yes you are right, many dynamic dependencies. Static linking of shared libs is not straight forward on Linux. The prebuilt binaries are compiled on Ubuntu 20.04.1 LTS. This version comes with ffmpeg 4.2.4.

**Reply by kwalker99** [reply]
cool, thanks for confirming

**Reply by apohl79** [reply]
I managed to link ffmpeg and libwebp statically on linux now. The problem were the ASM optimizations, which were breaking relocatable symbols. The next release will VST2/VST3 plugins with minimal list of dependencies to shared libs. This should improve cross distribution compatibility.
